src/laser_proc_nodelet.cpp

```diff
@@ -27,35 +27,36 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-/* 
+/*
  * Author: Chad Rockey
  */
 
-#include <laser_proc/LaserProcROS.h>
-#include <nodelet/nodelet.h>
+#include <memory>
+
+#include "rclcpp/node.hpp"
 
+#include "laser_proc/laser_proc_ros.hpp"
 
 namespace laser_proc
 {
-
-class LaserProcNodelet : public nodelet::Nodelet
+class LaserProcNodelet : public rclcpp::Node
 {
 public:
-  LaserProcNodelet()  {};
+  LaserProcNodelet()
+  : rclcpp::Node("laser_proc_nodelet")
+  {}
 
-  ~LaserProcNodelet() {}
+  ~LaserProcNodelet() = default;
 
 private:
   virtual void onInit()
   {
-    lp.reset(new LaserProcROS(getNodeHandle(), getPrivateNodeHandle()));
-  };
-  
-  boost::shared_ptr<LaserProcROS> lp;
-};
-
-}
+    lp.reset(new LaserProcROS(shared_from_this()));
+  }
```

Comment:
Is this `onInit()` method a leftover of ROS1 nodelets or where is this function actually being called?